__integrations__/models/message.test.ts

```diff
@@ -242,4 +262,290 @@ describe("Models |> Message", () => {
+        2 as NonNegativeInteger,
+        continuationToken as NonEmptyString
+      )
+      .map((ai: AsyncIterator<DecodedFeedResponse<RetrievedMessage>>) =>
+        ai.next().then(ir => ir.value)
+      )
+      .getOrElseL(_ => {
+        throw new Error("Error");
+      })
+      .run();
+
+    expect(results.length).toEqual(1);
+    results.forEach(i => {
+      expect(i.hasMoreResults).toBe(false);
+      expect(i.continuationToken).toBe(undefined);
+      expect(i.resource.isRight()).toBe(true);
+      // check that every message has a createdDate less than the previous one
+      if (i.resource.isRight()) {
+        expect(i.resource.value.createdAt < prevMessageCreationDate).toBe(true);
+        prevMessageCreationDate = i.resource.value.createdAt;
+      }
+    });
+
+    /* -------------------------------------------------------------------- */
+    /* let's now check that a newly started procedure will return message 4 */
+    /* -------------------------------------------------------------------- */
+
+    // get the first page of messages by fiscal code
+    results = await model
+      .findMessages(aFiscalCode, 2 as NonNegativeInteger)
+      .map((ai: AsyncIterator<DecodedFeedResponse<RetrievedMessage>>) =>
+        ai.next().then(ir => ir.value)
+      )
+      .getOrElseL(_ => {
+        throw new Error("Error");
+      })
+      .run();
+
+    prevMessageCreationDate = new Date();
+    continuationToken = "" as NonEmptyString;
+    expect(results.length).toEqual(2);
+    results.forEach(i => {
+      expect(i.hasMoreResults).toBe(true);
+      expect(i.continuationToken).not.toBe(undefined);
+      expect(i.resource.isRight()).toBe(true);
+      // check that every message has a createdDate less than the previous one
+      if (i.resource.isRight()) {
+        expect(i.resource.value.createdAt < prevMessageCreationDate).toBe(true);
+        prevMessageCreationDate = i.resource.value.createdAt;
+      }
+      continuationToken = i.continuationToken as NonEmptyString;
+    });
+
+    // get the second page of messages by fiscal code that should not have the message 4
+    results = await model
+      .findMessages(
+        aFiscalCode,
+        2 as NonNegativeInteger,
+        continuationToken as NonEmptyString
+      )
+      .map((ai: AsyncIterator<DecodedFeedResponse<RetrievedMessage>>) =>
+        ai.next().then(ir => ir.value)
+      )
+      .getOrElseL(_ => {
+        throw new Error("Error");
+      })
+      .run();
+
+    expect(results.length).toEqual(2);
+    results.forEach(i => {
+      expect(i.hasMoreResults).toBe(false);
+      expect(i.continuationToken).toBe(undefined);
+      expect(i.resource.isRight()).toBe(true);
+      // check that every message has a createdDate less than the previous one
+      if (i.resource.isRight()) {
+        expect(i.resource.value.createdAt < prevMessageCreationDate).toBe(true);
+        prevMessageCreationDate = i.resource.value.createdAt;
+      }
+    });
+  });
```

Comment:
We must manually dispose the created container after the test ends
```suggestion
    await context.dispose();
  });
```

Reply:
Done